sdk/identity/identity/src/credentials/managedIdentityCredential/appServiceMsi2017.ts

```diff
@@ -1,21 +1,29 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { createHttpHeaders, PipelineRequestOptions } from "@azure/core-rest-pipeline";
+import {
+  createHttpHeaders,
+  createPipelineRequest,
+  PipelineRequestOptions
+} from "@azure/core-rest-pipeline";
 import { AccessToken, GetTokenOptions } from "@azure/core-auth";
 import { credentialLogger } from "../../util/logging";
 import { MSI, MSIConfiguration } from "./models";
-import { mapScopesToResource, msiGenericGetToken } from "./utils";
+import { mapScopesToResource } from "./utils";
 
 const msiName = "ManagedIdentityCredential - AppServiceMSI 2017";
 const logger = credentialLogger(msiName);
 
+/**
+ * Formats the expiration date of the received token into the number of milliseconds between that date and midnight, January 1, 1970.
+ */
 function expiresInParser(requestBody: any): number {
```

Comment:
> This function is expiresIn but the property name is expires_on.

Changing it to `expiresOnParser` 👍

Reply:
Both `expires_in` and `expires_on` can exist in the payloads. Expires in meaning how much time is left, and expires on meaning at what time it will expire. We do this transformation on a case by case basis according to the target authentication environment. We try to narrow this down to the `expiresOnTimestamp` property, so I’ll change the `expiresInParser` functions to `expiresOnParser` since on average we use `expires.[oO]n` more often.

Reply:
If it's case by case, then I'm just saying that this specific case ought to be expiresOn. I noticed that one of the other MSI instances checks expires_in, but not this one and not several others.